theano/gpuarray/tests/test_subtensor.py

```diff
@@ -126,3 +128,26 @@ def test_adv_subtensor():
     rval = f(idx1_val, idx2_val)
     rep = xval[idx1_val, None, slice(0, 2, 1), idx2_val, None]
     assert numpy.allclose(rval, rep)
+
+
+class test_gpudiagonal(unittest.TestCase):
+    def test_matrix(self):
+        x = cuda.fmatrix()
+        np_x = numpy.arange(77).reshape(7, 11).astype('float32')
```

Comment:
I see no reason why this would not work for dtypes other than float32, so you might want to use config.floatX here.